File: src/main/java/semicontinuity/idea/code/analyzer/golang/QualifiedName.java
package semicontinuity.idea.code.analyzer.golang;

public class QualifiedName {
    private final String qualifier;
    private final String name;

    public QualifiedName(String qualifier, String name) {
        this.qualifier = qualifier;
        this.name = name;
    }

    public String getQualifier() {
        return qualifier;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return qualifier.isEmpty() ? name : qualifier + '.' + name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        QualifiedName that = (QualifiedName) o;

        if (!qualifier.equals(that.qualifier)) {
            return false;
        }
        return name.equals(that.name);
    }

    @Override
    public int hashCode() {
        int result = qualifier.hashCode();
        result = 31 * result + name.hashCode();
        return result;
    }
}

File: src/main/java/semicontinuity/idea/code/analyzer/golang/QualifiedName.kt
package semicontinuity.idea.code.analyzer.golang

class QualifiedName(val qualifier: String, val name: String) {
    override fun toString(): String {
        return if (qualifier.isEmpty()) name else "$qualifier.$name"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }

        val that = other as QualifiedName

        if (qualifier != that.qualifier) {
            return false
        }
        return name == that.name
    }

    override fun hashCode(): Int {
        var result = qualifier.hashCode()
        result = 31 * result + name.hashCode()
        return result
    }
}
